src/libponyrt/gc/serialise.c

```diff
@@ -1,15 +1,25 @@
+#ifdef __linux
+#define _GNU_SOURCE
```

Comment:
I don't think it is. On Linux, `RTLD_DEFAULT` is only available with GNU extensions enabled, but it looks available by default on FreeBSD.